File: modules/core/user/src/main/java/me/choicore/likeapuppy/core/user/jpa/entity/Identifier.java
package me.choicore.likeapuppy.core.user.jpa.entity;


import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Embeddable
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Identifier {
    @Column(unique = true, length = 50)
    private String email;

    @Column(unique = true, length = 20)
    private String mobile;

    public Identifier(
            final String email,
            final String mobile
    ) {
        this.email = email;
        this.mobile = mobile;
    }
}

File: modules/core/user/domain/src/main/kotlin/me/choicore/likeapuppy/core/domain/user/model/Identifier.kt
package me.choicore.likeapuppy.core.domain.user.model

import me.choicore.likeapuppy.core.domain.Validator

data class Identifier(
    val email: String,
    val phoneNumber: String? = null,
) : Validator {

    override fun validate() {
    }
}
